comp/core/tagger/impl/local_tagger.go

```diff
@@ -0,0 +1,185 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package taggerimpl
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/collectors"
+	taggercommon "github.com/DataDog/datadog-agent/comp/core/tagger/common"
+	tagger "github.com/DataDog/datadog-agent/comp/core/tagger/def"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/tagstore"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/telemetry"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	taggertypes "github.com/DataDog/datadog-agent/pkg/tagger/types"
+	"github.com/DataDog/datadog-agent/pkg/tagset"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+// Tagger is the entry class for entity tagging. It hold the tagger collector,
+// memory store, and handles the query logic. One should use the package
+// methods in comp/core/tagger to use the default Tagger instead of instantiating it
+// directly.
+type localTagger struct {
+	sync.RWMutex
+
+	tagStore      *tagstore.TagStore
+	workloadStore workloadmeta.Component
+	cfg           config.Component
+	collector     *collectors.WorkloadMetaCollector
+
+	ctx            context.Context
+	cancel         context.CancelFunc
+	telemetryStore *telemetry.Store
+}
+
+func newLocalTagger(cfg config.Component, workloadStore optional.Option[workloadmeta.Component], telemetryStore *telemetry.Store) (tagger.Component, error) {
+	wmeta, ok := workloadStore.Get()
+	if !ok {
+		return nil, fmt.Errorf("workloadmeta component is not available")
```

Comment:
It is an optional 😄 

Am I missing something?

Reply:
`workloadStore` type was changed from `workloadmeta.Component` to `optional.Option[workloadmeta.Component]`. At the same time, if `workloadStore` is `None`, `newLocalTagger` returns an error. So it seems `workloadStore` is not optional but required. Should the type of  `workloadStore` be `workloadmeta.Component`? Did I miss something?

Reply:
This is mostly to ensure we can use the `fx-dual` and have the same dependencies. Since remote uses optional and the local tagger uses non-optional, I changed the dependency for local tagger to be optional, but fail if the workloadmeta is optional. This is kind of an edge case because we support using remote or local and we decide at runtime.